soccer/gameplay/plays/testing/test_adaptive_formation_weights.py

```diff
@@ -0,0 +1,126 @@
+import main
+import play
+import behavior
+import evaluation.field
+import evaluation.passing_positioning
+import evaluation.shooting
+import evaluation.defensive_positioning
+import constants
+import robocup
+import math
+import enum
+
+
+class TestAdaptiveFormationWeights(play.Play):
```

Comment:
I'm going to leave this for another issue so this one doesn't get /too/ complicated. I should be able to implemented it later this week / next week depending on my schedule.